### PR TITLE
Bloom sharding

### LINKED ISSUE #139: Reduce Bloom Filter Sizes
Bloom filters can be multiple MBs, result in our largest GCS accesses and will be difficult to cache.

Review options for bitmap compression.  This came up recently:
http://roaringbitmap.org/about/

Review options and consider things like compression ratio and ability to retrieve in a paged manner: #32.

### PR DESCRIPTION
**Pull request by annanay25**
Bloom filter sharding is implemented to reduce bloom sizes for larger blocks, this makes them easily cacheable. Slight but not significant reduction in false positive rate is also expected as there is another round of hashing involved.

Fixes #139

### CODE REVIEW

File: tempodb/backend/util/util.go [hunk 1]
```diff
@@ -3,35 +3,37 @@ package util
...
 
 func MetaFileName(blockID uuid.UUID, tenantID string) string {
-	return path.Join(rootPath(blockID, tenantID), "meta.json")
+	return path.Join(RootPath(blockID, tenantID), "meta.json")
 }
 
-func BloomFileName(blockID uuid.UUID, tenantID string) string {
-	return path.Join(rootPath(blockID, tenantID), "bloom")
+func BloomFileName(blockID uuid.UUID, tenantID string, bloomShard int) string {
+	return path.Join(RootPath(blockID, tenantID), "bloom-"+strconv.Itoa(bloomShard))
```
Comment (mdisibio) [inline tempodb/backend/util/util.go:16]: Should we consider backwards compatibility here?  I.e. reading the existing "bloom" file if present.

Comment (joe-elliott) [reply tempodb/backend/util/util.go:16]: I think for the time being we're fine with breaking changes.  There will definitely come a point when we will make sure storage is always backwards compatible.

File: tempodb/tempodb.go [hunk 2]
```diff
@@ -253,21 +252,26 @@ func (rw *readerWriter) Find(ctx context.Context, tenantID string, id encoding.I
 			copiedBlocklist = append(copiedBlocklist, b)
 		}
 	}
+	numTenants := len(rw.blockLists)
 	rw.blockListsMtx.Unlock()
 
-	if !found {
+	if numTenants == 0 {
```
Comment (joe-elliott) [inline tempodb/tempodb.go:258]: let's leave this out for now.  this is a different issue and @mdisibio is going to be looking at it.

**Review by joe-elliott — CHANGES_REQUESTED** [review]
- One small thing to remove
- Add a changelog entry and note it is a breaking change

let's get this merged :)

File: tempodb/tempodb.go (hunk 2 above)
Comment (annanay25) [reply tempodb/tempodb.go]: This should fix https://github.com/grafana/tempo/issues/94 but I'll defer to @mdisibio for the fix!

**Review by joe-elliott — APPROVED** [review]
lgtm

**Pull request merged**